lldb/source/Core/Statusline.cpp

```diff
@@ -102,20 +103,36 @@ void Statusline::UpdateScrollWindow(ScrollWindowMode mode) {
   const unsigned scroll_height =
       (mode == DisableStatusline) ? m_terminal_height : m_terminal_height - 1;
 
+  CursorPosition cursor_position = m_debugger.GetIOHandlerCursorPosition();
```

Comment:
Move this into the if, I know it has to be done before the newline, but it can still be closer to point of use.